docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,284 @@
+Backend Configuration
+=====================
+
+NeuroConv offers highly convenient control over the type of file backend and the way its datasets are configured.
+
+Find out more about possible backend formats in the `main NWB documentation <https://nwb-overview.readthedocs.io/en/latest/faq_details/why_hdf5.html#why-use-hdf5-as-the-primary-backend-for-nwb>`_.
+
+Find out more about chunking and compression in the `advanced NWB tutorials for dataset I/O settings <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/h5dataio.html#sphx-glr-tutorials-advanced-io-h5dataio-py>`_.
+
+Find out more about memory buffering of large source files in the `advanced NWB tutorials for iterative data write <https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/plot_iterative_write.html#sphx-glr-tutorials-advanced-io-plot-iterative-write-py>`_.
+
+
+
+Default configuration
+---------------------
+
+To retrieve a default configuration for an in-memory ``pynwb.NWBFile`` object, use the ``get_default_backend_configuration`` function...
+
+.. code-block:: python
+
+    from datetime import datetime
+    from uuid import uuid4
+
+    from neuroconv.tools.nwb_helpers import get_default_backend_configuration
+    from pynwb import NWBFile, TimeSeries
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0)
+    nwbfile = NWBFile(
+        identifier=str(uuid4()),
+        session_start_time=session_start_time,
+        session_description="A session of my experiment.",
+    )
+
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    default_backend_configuration = get_default_backend_configuration(
+        nwbfile=nwbfile, backend="hdf5"
+    )
+
+From which a printout of the contents looks like...
+
+.. code-block:: python
+
+    print(default_backend_configuration)
+
+.. code-block:: bash
+
+    Configurable datasets identified using the hdf5 backend
+    -------------------------------------------------------
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
```

Comment:
```suggestion
    ===========================
```

minor point, but if we are going to use RST syntax for headers we might as well format it so that it gets rendered properly in RST

Reply:
This line is in the python object printout in console - the `print` is not intended to be RST format, the dashes are just a way of visually separating the datasets and their contents (similar to numpy docstring style, like what would be printed from `?` or `help`)

I gave it a try out of curiosity, but since it is a code block, RST rules do not apply

![image](https://github.com/catalystneuro/neuroconv/assets/51133164/31f9d694-ed21-495c-9ed5-cd123da63ba4)

We could take it out of the code block to apply the headers as you say, but then the appropriate header characters would also be relative to _where_ in the docs (at what level of indentation) the printout is meant to be placed

Reply:
that's ok we can just leave it as is